### PR TITLE
Fix env variables and sentry source upload

### LINKED ISSUE #399: Generate and upload sourcemaps to Sentry

### PR DESCRIPTION
**Pull request by mpetrunic**
- add env variable replacement during build
- add sentry plugin for sourcemap upload
- remove type and eslint checking in webpack

@morrigan Can you take a detail look and set required secrets.

resolves #399

### CODE REVIEW

**Comment by morrigan** [comment]
I've already set the secrets because using them in my draft PR so check to align that.

**Comment by mpetrunic** [comment]
> I've already set the secrets because using them in my draft PR so check to align that.

there is only sentry dns from what I saw

File: webpack.base.config.js [hunk 1]
```diff
@@ -30,6 +30,9 @@ module.exports = {
         new Dotenv(),
         new webpack.IgnorePlugin({
             resourceRegExp: /\@chainsafe\/blst/,
-          })
+          }),
+          new webpack.DefinePlugin({
+            "process.env.DOCKER_LIGHTHOUSE_IMAGE": JSON.stringify(process.env.DOCKER_LIGHTHOUSE_IMAGE || "sigp/lighthouse:v1.0.5")
```
Comment (BeroBurny) [inline webpack.base.config.js:35]: better use `sigp/lighthouse:latest` verzion `1.0.5` could be deprecated in few months

**Comment by BeroBurny** [comment]
probably this package will help in maintenance https://www.npmjs.com/package/dotenv-webpack

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -110,6 +110,11 @@ jobs:
           mac_certs_password: ${{ secrets.MAC_PASSWORD }}
         env:
           EP_PRE_RELEASE: true
+          SENTRY_DSN: ${{ secrets.SENTRY_DSN }}
+          SENTRY_ORG: "nodefactory"
+          SENTRY_PROJECT: "chainguardian"
+          SENTRY_SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
```
Comment (morrigan) [inline .github/workflows/release.yaml:116]: this is not used, can be removed

File: webpack.base.prod.config.js [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
+        "process.env.SENTRY_AUTH_TOKEN": JSON.stringify(process.env.SENTRY_AUTH_TOKEN || ""),
+    }),
+];
+
+if(process.env.SENTRY_AUTH_TOKEN) {
+    // Sentry CLI used for uploading releases
+    plugins.push(new SentryWebpackPlugin({
+
+        // sentry-cli configuration
+        authToken: process.env.SENTRY_AUTH_TOKEN,
```
Comment (morrigan) [inline webpack.base.prod.config.js:19]: I see now. You replace DSN usage with this?

Comment (mpetrunic) [reply webpack.base.prod.config.js:19]: this is for sourcemap upload (private value), dsn is for error reporting (public value)

File: .github/workflows/release.yaml (hunk 2 above)
Comment (mpetrunic) [reply .github/workflows/release.yaml:116]: it's used by webpack during release process to upload sourcemap

**Review by morrigan — APPROVED** [review]
Looks good. Added auth secret

File: webpack.base.config.js (hunk 1 above)
Comment (morrigan) [reply webpack.base.config.js:35]: ha, double edged sword :D
We can let users do "docker pull" if update is needed. To me looks better to be on stable version that's working for us

**Pull request merged**